_pytest/unittest.py

```diff
@@ -152,14 +151,33 @@ def addSuccess(self, testcase):
     def stopTest(self, testcase):
         pass
 
+    def _handle_skip(self):
+        # implements the skipping machinery (see #2137)
+        # analog to pythons Lib/unittest/case.py:run
+        testMethod = getattr(self._testcase, self._testcase._testMethodName)
+        if (getattr(self._testcase.__class__, "__unittest_skip__", False) or
+            getattr(testMethod, "__unittest_skip__", False)):
+            # If the class or method was skipped.
+            skip_why = (getattr(self._testcase.__class__, '__unittest_skip_why__', '') or
+                        getattr(testMethod, '__unittest_skip_why__', ''))
+            try:  # PY3, unittest2 on PY2
+                self._testcase._addSkip(self, self._testcase, skip_why)
```

Comment:
Thanks @mbyt, now I see that it is support both `unittest` and `unittest2` on Python 2.

A try/except for TypeError still makes me uneasy, but I have no better suggestion so let's go with it. 👍